src/psyclone/psyir/nodes/psy_data_node.py

```diff
@@ -872,15 +676,15 @@ def gen_type_bound_call(typename, methodname, argument_list=None,
                     module_name = f"@{module_name}"
                 call = gen_type_bound_call(
                     self._var_name, "PreDeclareVariable",
-                    [f"\"{sig}{module_name}{pre_suffix}\"", unique_sig])
+                    [f"\"{sig}{pre_suffix}{module_name}\"", unique_sig])
```

Comment:
A 'suffix' should be appended, not prepended. What's the reason for this change?

Reply:
@hiker suggested it to pass the tests. Was it maybe reverted between gen_code and lowering?